recovery/src/main/AndroidManifest.xml

```diff
@@ -3,14 +3,14 @@
 
     <application>
         <provider
-            android:authorities="com.kin.ecosystem.recovery"
+            android:authorities="${applicationId}.KinRecoveryFileProvider"
             android:exported="false"
             android:grantUriPermissions="true"
             android:name="android.support.v4.content.FileProvider">
```

Comment:
In the end, you didn't change android:name - I don't think we need to just to make sure it on purpose?